src/prezzemolo/graph.py

```diff
@@ -87,6 +110,42 @@ def breadth_first_search(self, start: Vertex[ValueType], end: Vertex[ValueType])
         # start and end are type checked inside _breadth_first_search()
         return self._breadth_first_search(start, end, vertex_2_parent=None)
 
+    def _dijkstra_search(
+        self, start: Vertex[ValueType], end: Optional[Vertex[ValueType]], vertex_2_parent: Optional[Dict[Vertex[ValueType], Optional[Vertex[ValueType]]]] = None
+    ) -> bool:
+        distance: Dict[Vertex[ValueType], float] = {v: float("inf") for v in self.__vertexes}
+        distance[start] = 0.0
+        remaining_vertexes: PriorityQueue[_ShortestDistance[ValueType]] = PriorityQueue()
+        remaining_vertexes.put(_ShortestDistance(distance=0.0, vertex=start))
+        visited: Set[Vertex[ValueType]] = set()
+
+        while not remaining_vertexes.empty():
+            next_vertex: _ShortestDistance[ValueType] = remaining_vertexes.get()
+            current_distance: float = next_vertex.distance
+            current_vertex: Vertex[ValueType] = next_vertex.vertex
+
+            if current_vertex in visited:
+                continue
+
+            visited.add(current_vertex)
+
+            for neighbor in current_vertex.neighbors:
+                neighbor_distance = current_distance + current_vertex.get_weight(neighbor)
+                if neighbor_distance < distance[neighbor]:
+                    if vertex_2_parent is not None:
+                        vertex_2_parent[neighbor] = current_vertex
+                    distance[neighbor] = neighbor_distance
+                    remaining_vertexes.put(_ShortestDistance(distance=neighbor_distance, vertex=neighbor))
```

Comment:
No need to put anything into `remaining_vertexes` if we initialize it as described above.

Reply:
This was my mistake: ignore this.